module-2/backend/week_three.md

```diff
@@ -10,14 +10,62 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What is the entry at the command line to create a new rails app?
+<br>
+rails new database=postgresql -T
+<br>
 2. What do Models generally inherit from in rails?
+<br>
+They inherit form ActiveRecord::Base
+<br>
 3. What do Controllers generally inherit from in a rails project?
+<br>
+They generally inherit from the ApplicationsController
+<br>
 4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+<br>
+get "/horses/:id" to: "horses#show"
+<br>
 5. What rake task is useful when looking at routes, and what information does it give you?
+<br>
+Rake routes give the user a list of all existing routes in the terminal.
+<br>
 6. What is an example of a route helper? When would you use them?
+<br>
+Route helpers allow the user to create paths without explicitly stating the url e.g. horses_path vs /horses.
+<br>
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+<br>
+The url is the actual address that will appear for the user while the path is used to navigate the site internally.
+<br>
 8. What are strong params and why are the necessary?
+<br>
+Strong params protect the attributes from user assignment, only allowing assignment to take place if they have been listed in strong_params.
+<br>
 9. What role does `form_for` play in helping us create our forms?
+<br>
+Form_for is a shortcut that allows us to quickly create forms in rails that usually take in information regarding objects.
+<br>
 10. How does `form_for` know where to submit the user's input?
-11. Create a form using a `form_for` helper to create a new `Horse`. 
+<br>
+Form_for knows where to submit the users input by looking at where the request to it came from.
```

Comment:
It actually looks at the argument that's passed to it (e.g. `@horse` in the example below), and checks to see if that argument has been saved to the DB or not. If not, it creates a form that submits a POST request to `/horses`. If so, it creates a form that submits a PATCH request to `/horses/:id`.